lib/Logger/LogThread.cpp

```diff
@@ -32,7 +32,7 @@ arangodb::basics::ConditionVariable* LogThread::CONDITION = nullptr;
 boost::lockfree::queue<LogMessage*>* LogThread::MESSAGES = nullptr;
 
 LogThread::LogThread(application_features::ApplicationServer& server, std::string const& name)
-    : Thread(server, name), _messages(0) {
+    : Thread(server, name), _messages(16384) {
```

Comment:
I am not sure if this change is justified.
The boost::lockfree::queue we are using _can_ already grow dynamically. So I think we can get away with a much smaller reserve size, e.g. 32 or 64, and then have it grow dynamically if it actually needs more space (it will never do in most environments).
```suggestion
    : Thread(server, name), _messages(16384) {
```